src/config.js

```diff
@@ -1,16 +1,82 @@
-export const WMTS_BASE_URL = process.env.VUE_APP_WMTS_BASE_URL
-export const WMS_BASE_URL = process.env.VUE_APP_WMS_BASE_URL
+/**
+ * Flag that tells if the app is currently in debug mode (a.k.a. built in dev mode)
+ * @type Boolean
+ */
+export const DEBUG = process.env.VUE_APP_DEBUG
 
-// comes from https://github.com/geoadmin/mf-geoadmin3/blob/master/src/components/map/TileGrid.js
+/**
+ * Flag that tells if the app is currently running in a Cypress environment for E2E testing
+ * @type Boolean
+ */
+export const IS_TESTING_WITH_CYPRESS = !!window.Cypress
+
+/**
+ * Adds a slash at the end of the URL if there is none
+ * @param {String} url
+ * @return {String} the URL with a trailing slash
+ */
+function enforceEndingSlashInUrl(url) {
+  if (url && !url.endsWith('/')) {
+    return `${url}/`
+  }
+  return url
+}
+
+/**
+ * Base part of the URL to use when requesting the backend (e.g. for prod https://api3.geo.admin.ch).
+ *
+ * This URL always end with a slash, so there's no need at add another one after it to create REST endpoints
+ * @type String
+ */
+export const API_BASE_URL = enforceEndingSlashInUrl(process.env.VUE_APP_API_BASE_URL)
+/**
+ * Base part of the URL to use when requesting GeoJSON data (e.g. for prod https://data.geo.admin.ch).
+ *
+ * This URL always end with a slash, so there's no need at add another one after it to create REST endpoints
+ * @type {String}
+ */
+export const DATA_BASE_URL = enforceEndingSlashInUrl(process.env.VUE_APP_DATA_BASE_URL)
+/**
+ * Default WMTS base part of the URL to use when requesting tiles (e.g. for prod https://wmts{5-9}.geo.admin.ch/).
+ *
+ * This URL always end with a slash, so there's no need at add another one after it to create REST endpoints
+ * @type {String}
+ */
+export const WMTS_BASE_URL = enforceEndingSlashInUrl(process.env.VUE_APP_WMTS_BASE_URL)
+/**
+ * Default WMS base part of the URL to use when requesting tiles (e.g. for prod https://wms{0-4}.geo.admin.ch).
+ *
+ * This URL always end with a slash, so there's no need at add another one after it to create REST endpoints
+ *
+ * WMS layers tend to carry their own base URL in their metadata, be mindful of that when using this constant (it might be unnecessary)
+ * @type {String}
+ */
+export const WMS_BASE_URL = enforceEndingSlashInUrl(process.env.VUE_APP_WMS_BASE_URL)
+
+/**
+ * Default tile size to use when requesting WMS tiles with our internal WMSs (512px)
+ *
+ * comes from {@link https://github.com/geoadmin/mf-geoadmin3/blob/master/src/components/map/TileGrid.js}
+ * @type {Number}
+ */
 export const WMS_TILE_SIZE = 512 // px
 
-// stuff that comes from https://github.com/geoadmin/mf-geoadmin3/blob/master/mk/config.mk
-export const TILEGRID_ORIGIN = [558147.8, 6152731.53] // [2420000, 1350000] projected from EPSG:2056 to EPSG:3857
-// resolutions in meter/pixel
+/**
+ * Origin of the TileGrid (comes from {@link https://github.com/geoadmin/mf-geoadmin3/blob/master/mk/config.mk}) reprojected in EPSG:3857
```

Comment:
for newbie like me, it would be good to have the meaning of EPSG:3857 in parenthese (WGS84)  :-)